### ISSUE TITLE
Campaign marked as finished after pause/restart/resume

### ISSUE THREAD

**Issue by vixns** [knadh/listmonk#2168]
I changed the sliding window limit while a campaign was active.

Then I follow the instruction, pause all active campaigns and restart

After restart, the campaign was marked as finished immediately after resuming, but the progress is `Sent 5,258 / 5,601`

This campaign was sent "later", scheduling allows pause/resume action ?

Is there a sql/manual way to resume this campaign ?

listmonk v4.1.0 deployed by argocd on k8s

**Comment by mevinbabuc** [comment]
Yup, this issue is still there. I'm trying to figure out how to restart this manually by updating the tables. Unable to figure out which subscribers where already sent emails.

**Comment by knadh** [comment]
I tried changing various performance settings while a campaign is running (with 5000 dummy subscribers and a local smtp). Also tried pausing/resuming a campaign in rapid succession. Tried a combination of these things, but unable to reproduce this scenario.

Would be helpful if the exact steps that allow this issue to be reproduced is described.

**Comment by thewaywest** [comment]
Hello, @knadh ~

I just hit this issue with a new campaign, and followed the link to your Apr 24 patch. Github says it is not connected to any branch in the project -- has it been implemented? Or perhaps there is another workaround for the problem?

- Listmonk 5.1.0, docker stack
- Using Resend.com service for relay: free tier permits 100 mails/day and 2 mails/sec
- Performance settings modified to reflect these constraints:
  - Concurrency = 1
  - Rate = 1
  - Batch = 1000
  - Max Error = 1 (testing system, want things to stop before too many recipients are hit)
  - Sliding Window = true
    - Max messages = 100
    - Duration = 24h

No errors were reported in the log. The session unpaused with 90/205 messages sent.

> ⇨ http server started on 10.8.0.113:9000
> 2025/10/22 13:36:17.673724 manager.go:423: start processing campaign (Sangha of the Falling Rain | Sharing the Merit)  
> 2025/10/22 13:36:17.691648 pipe.go:232: campaign (Sangha of the Falling Rain | Sharing the Merit) finished  
> 2025/10/22 13:44:17.768206 init.go:963: reloading on signal ...  
> 2025/10/22 13:44:17.768563 init.go:904: HTTP server shut down

Web search results suggested there may be an issue describing the window in hours; however, the behavior (finishing instantly) looks more like the above.

If this has elements that are unrelated to the this issue, I am happy to create a new one.

Thank you for your work!

**Comment by rikmeijer** [comment]
Did you figure it out? I am running into this problem after a forced server reboot

> Yup, this issue is still there. I'm trying to figure out how to restart this manually by updating the tables. Unable to figure out which subscribers where already sent emails.
